File: documentation-examples-java/src/java/dev/typr/foundations/docs/postgresql/NullableTypes.java
package dev.typr.foundations.docs.postgresql;

import dev.typr.foundations.PgType;
import dev.typr.foundations.PgTypes;
import java.util.Optional;

@SuppressWarnings("unused")
public class NullableTypes {
    //start
    PgType<Integer> notNull = PgTypes.int4;
    PgType<Optional<Integer>> nullable = PgTypes.int4.opt();  // null values allowed
    //stop
}

File: documentation-examples-kotlin/src/kotlin/dev/typr/foundationskt/docs/postgresql/NullableTypes.kt
package dev.typr.foundationskt.docs.postgresql

import dev.typr.foundationskt.*
import dev.typr.foundationskt.data.*

@Suppress("unused")
class NullableTypes {
    //start
    val notNull: PgType<Int> = PgTypes.int4
    val nullable: PgType<Int?> = PgTypes.int4.opt()
    //stop
}
